modules/acpica/CMakeLists.txt

```diff
@@ -0,0 +1,177 @@
+# Copyright (c) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+set(ACPI_DIR ${ZEPHYR_CURRENT_MODULE_DIR}/)
+set(INC_DIR ${ACPI_DIR}/source/include/)
+set(SRC_DIR ${ACPI_DIR}/source)
+set(COMP_DIR ${ACPI_DIR}/source/components)
+set(PARENT_SRC_DIR ${ACPI_DIR}../../zephyr)
+set(ACPI_PARENT_DIR ${ACPI_DIR}/../)
+
+zephyr_include_directories(
+	${PARENT_SRC_DIR}/include/
+	${ACPI_PARENT_DIR}/
```

Comment:
This is the same as `modules/lib` in the west project directory hierarchy, right? I do think it's an improvement that you have some directory namespacing for the ACPICA headers, but I'm not sure it's ideal that we add a new include lookup path for a directory that's this generic (not specific to acpica). That said, it might be good enough for an initial solution, so we get this merged and can improve the situation in subsequent PRs.